pp_ctl.c: comment on unreachable code

jkeenan opened the pull request:
Addresses https://github.com/Perl/perl5/issues/23964.

-------------------------------------------------------------------------------
* This set of changes does not require a perldelta entry.

iabyn reviewed (commented):
Given that its not slowing anything down by being there, I would be tempted to keep in case of any hypothetical XS-constructed optrees which bypass the compile-time check.

jkeenan committed:
pp_ctl.c: Retain code but explain why we're doing so

Test suite does not reach this code block, but Dave M recommends
retaining it in case of (hypothetical) XS-constructed optrees
which bypass compile-time check.

For: GH #23964 (issue); GH #23965 (p.r.)

tonycoz commented on pp_ctl.c:
```diff
@@ -3596,6 +3596,9 @@ PP(pp_goto)
                 /* diag_listed_as: Can't "%s" out of a "defer" block */
+                /* GH #23965: Retain in case of any hypothetical
+                 * XS-constructed optrees which bypass the compile-time check.
+                 * */
```
While you say why we might keep it, you don't say why it was considered for removal.

jkeenan replied:
I've added to the inline comment.  Is that what you were looking for?

tonycoz replied:
Something like that, though the code is reachable so this PR is obsolete.

jkeenan committed:
pp_ctl.c pp_goto: Elaborate on rationale in comment
